### PR TITLE
Added functionality to open openHAB Preview on macOS based systems

### LINKED ISSUE #15: Use openHAB preview on macOS
The logic for building the address for the preview in VSCode is semi static and only works on windows.

I made a quick fix in a PR. #14 

## Expected Behavior
openHAB preview works on macOS

## Current Behavior
The extension tries to parse the files path under macOS and fails. No output.

## Env
macOS

### PR DESCRIPTION
**Pull request by dennisausbremen**
Unfortunately I'm not able to test on Linux atm.
Since there seems to be no "configurability" in VSCode extensions a lot has to be hardcoded. :disappointed:

Closes #15

### CODE REVIEW

**Comment by kubawolanin** [comment]
Hi @dennisausbremen! Thank you for your contribution.

> Since there seems to be no "configurability" in VSCode extensions a lot has to be hardcoded

In fact, there is a way to provide configuration points for the extension. Please take a look:

https://code.visualstudio.com/docs/extensionAPI/extension-points#_contributesconfiguration

All we need to do is to add a "configuration" node within "contributes" in `package.json`:
https://github.com/openhab/openhab-vscode/blob/master/package.json#L27

```json
    "configuration": {
        "type": "object",
        "title": "openHAB configuration",
        "properties": {
            "openhab.hostname": {
                "type": "string",
                "default": null, // or 'localhost' ?
                "description": "Hostname or IP address of openHAB instance."
            }
        }
    }
```

And then in the code you would just use this config like this:
```js
let config = vscode.workspace.getConfiguration('openhab')
console.log(config.hostname) // localhost or openhabianpi etc.
```

A few remarks, though:

* We'll need to parse the value in case user provides it with protocol and/or port, e.g. `http://home` or `home:8080`. - What we need is `home` only :-) We will reuse that in further features, like REST connection mentioned in #7 
* [README.md](https://github.com/openhab/openhab-vscode/blob/master/README.md#configuration) file would need to change as well as we're introducing a configuration point here.

Thanks again!

File: src/extension.ts [hunk 1]
```diff
@@ -55,12 +55,27 @@ async function init(context: ExtensionContext, disposables: Disposable[]): Promi
...
+                return _.compact(filePath)[0]
+            } else if ( os === 'darwin' && absolutePath.slice(0,9) === '/Volumes/') {
+                return 'openhabianpi.local'
+            } else {
+                return 'localhost'
+            }
+        }
+
+        const getFileName = () => {
+            return (os === 'win32') ? absolutePath.split('\\').pop() : absolutePath.split('/').pop()
```
Comment (kubawolanin) [inline src/extension.ts]: Just a thought. We could utilize [Node.js path library](http://devdocs.io/node/path#path_path_join_paths) so it does the detection for us :)

Comment (dennisausbremen) [inline src/extension.ts]: Yes, it was late yesterday. 😄
Will do. 
Thanks for the input! 🎉

**Comment by dennisausbremen** [comment]
> In fact, there is a way to provide configuration points for the extension. Please take a look
I think i just missed that. Great. 😆 👍 

The main problem with the parsing of the correct path is that it just works somewhat great on windows. 😞 
Maybe we should abandon the auto-detection alltogether or make it optional or detect if the user is using "win32" to automatically turn it on, since it's of no real use on a mac or *NIX machine.

> We'll need to parse the value in case user provides it with protocol and/or port
Using path() that should be no problem. 😉 

> README.md file would need to change as well as we're introducing a configuration point here.
I agree. 👍

**Comment by kubawolanin** [comment]
> Maybe we should abandon the auto-detection alltogether

I'm perfectly fine with this. Let's just make sure that `hostname` provided in the user settings could be reusable.

Cheers! :smiley:

**Comment by dennisausbremen** [comment]
Updated my code to accomodate your feedback.

File: README.md [hunk 2]
```diff
@@ -18,8 +18,36 @@ The extension is designed with openHAB 2.x in mind - most snippets and design pa
...
 ## Configuration
 
-You don't need to provide any configuration to this extension. 
-It automatically detects the hostname of openHAB. Detection is based on the path of active file.
+You are able to configure the hostname and port for the Sitemap preview.
+
+* openhab.host (mandatory), default: openhabianpi
+* openhab.port (optional), default: 8080
+
+These settings should work fine on Windows machines and openHAB installations using the recommended [openHABian](http://docs.openhab.org/installation/openhabian.html) setup.
```
Comment (kubawolanin) [inline README.md]: +1 for openHABian recognition! :-)

**Comment by kubawolanin** [comment]
@dennisausbremen your changes LGTM overall. There's one thing that's needed from you before I merge it.

Can you squash your commits and change commit message to include a Signed-off-by (to certify that you wrote it or otherwise have the right to pass it on as an open-source), see http://docs.openhab.org/developers/contributing/contributing.html#sign-your-work

For squashing / rebasing see:
http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

Also, if you edit your first comment with `Closes #15` annotation, it'll automatically refer to and close the issue you've created. Thanks!

**Comment by dennisausbremen** [comment]
@kubawolanin: Like this? 😄

**Comment by kubawolanin** [comment]
*perfecto!* Thanks again mate!

**Comment by dennisausbremen** [comment]
No problem! 🍸😎👍

File: README.md (hunk 2 above)
Comment (ThomDietrich) [inline README.md]: +1 Indeed :smile:

**Pull request merged**